src/gdf_table.cuh

```diff
@@ -0,0 +1,321 @@
+            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
+            if(my_elem != other_elem)
+              return false;
+            break;
+          }
+        case GDF_INT64:
+          {
+            using col_type = int64_t;
+            const col_type my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
+            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
+            if(my_elem != other_elem)
+              return false;
+            break;
+          }
+        case GDF_FLOAT32:
+          {
+            using col_type = float;
+            const col_type my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
+            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
+            if(my_elem != other_elem)
+              return false;
+            break;
+          }
+        case GDF_FLOAT64:
+          {
+            using col_type = double;
+            const col_type my_elem = static_cast<col_type*>(d_columns_data[i])[my_row_index];
+            const col_type other_elem = static_cast<col_type*>(other.d_columns_data[i])[other_row_index];
+            if(my_elem != other_elem)
+              return false;
+            break;
+          }
+        default:
+          printf("Attempted to compare columns of unsupported GDF datatype\n");
+          return false;
+      }
+    }
+
+    return true;
+  }
+
+  /* --------------------------------------------------------------------------*/
+  /** 
+   * @Synopsis  This device function computes a hash value for a given row in the table
+   * 
+   * @Param row_index The row of the table to compute the hash value for
+   * @Param num_columns_to_hash The number of columns in the row to hash. If 0, hashes all columns
+   * @tparam hash_function The hash function that is used for each element in the row
+   * @tparam dummy Used only to be able to resolve the result_type from the hash_function.
+                   The actual type of dummy doesn't matter.
+   * 
+   * @Returns The hash value of the row
+   */
+  /* ----------------------------------------------------------------------------*/
+  template <template <typename> class hash_function = default_hash,
+            typename dummy = int>
+  __device__ 
+  typename hash_function<dummy>::result_type hash_row(size_type row_index, 
+                                                      size_type num_columns_to_hash = 0) const
+  {
+    using hash_value_t = typename hash_function<dummy>::result_type;
+    hash_value_t hash_value{0};
+
+    // If num_columns_to_hash is zero, hash all columns
+    if(0 == num_columns_to_hash)
+      num_columns_to_hash = this->num_columns;
+
+    for(size_type i = 0; i < num_columns_to_hash; ++i)
+    {
+      const gdf_dtype current_column_type = d_columns_types[i];
+
+      switch(current_column_type)
+      {
+        case GDF_INT8:
```

Comment:
there are missing data types here. There is an api i think in functions.h or types.h I can't remember that gives you the width of a gdf_column per value in bytes. You could use this width and just cast the pointer to one of the types you have implemented here. If you are hashing a float64 or an int64 the operation is exactly the same. 

This would make it so that when people add new data types you would not have to add new if statements here.